elastic/apm-server
# RUM: Route transaction|span events on agent.name

## Motivation/summary

When datastreams are enabled, the RUM spans and transactions are sent to
the `traces-apm.rum` datastream. The supported RUM Agent names are:
    - "js-base" (Legacy RUM agent)
    - "rum-js" (RUM agent)
    - "iOS/swift" (iOS Agent)

An ILM Delete phase has been set up per datastream as follows:

|        Data Stream     | Delete Phase Days |
| ---------------------- | ----------------- |
| `logs-apm.error`       | `10d`             |
| `metrics-apm.profiling`| `10d`             |
| `traces-apm`           | `10d`             |
| `traces-apm.rum`       | `90d`             |
| `metrics-apm.app`      | `90d`             |
| `metrics-apm.internal` | `90d`             |

## Checklist


## How to test these changes

1. `docker-compose up -d`
2. `make`
3. `./apm-server -E apm-server.rum.enabled=true -E apm-server.data_streams.enabled=true -E output.elasticsearch.username=admin -E output.elasticsearch.password=changeme`
4. `curl -H "Content-type: application/x-ndjson" --data-binary @testdata/intake-v3/rum_events.ndjson http://localhost:8200/intake/v3/rum/events`
5. `curl -H "Content-type: application/x-ndjson" --data-binary @testdata/intake-v2/transactions_spans_rum_2.ndjson http://localhost:8200/intake/v2/rum/events`
6. Navigate to Kibana Dev Tools and issue the following query and verify: 

```javascript
GET traces-apm.rum*/_search
{
  "_source": [false], 
  "fields": [
    "processor.*",
    "data_stream.dataset",
    "agent.name"
  ], 
  "query": {
    "match_all": {}
  }
}
```
  - `data_stream.dataset == apm.rum`
  - `processor.event == transaction|span`
  - `agent.name == rum-js|js-base`


6. Verify that the transactions show up in `Observability > APM > Services`. (There should be 2 services with 1 transaction each).
7. Verify that the ILM delete phases are set up in ILM.

## Related issues

Closes #6223
